aten/src/ATen/core/OptionsGuard.h

```diff
@@ -9,28 +9,53 @@
 
 namespace at {
 
-// In the CAFFE2_FB_LIMITED_MOBILE_CAPABILITY build setting,
-// thread_local is not supported.  In that case, we don't provide
-// an OptionsGuard; and force you to pass around options manually.
-#if !AT_MOBILE && !defined(CAFFE2_FB_LIMITED_MOBILE_CAPABILITY)
-
 /// A wrapper over a thread local TensorOptions instance.
+/// INVARIANT: all fields are NOT nullopt
 struct DefaultTensorOptions {
-  /// Returns the current thread local default options.
-  /// Defined in OptionsGuard.cpp because we can't use optional in headers, due
-  /// to Windows and other compilers.
-  /// TODO: The inability to use optional in headers is no longer true
-  CAFFE2_API static TensorOptions& get();
+  DefaultTensorOptions() {}
+  DefaultTensorOptions(const DefaultTensorOptions&) = default;
+  DefaultTensorOptions& operator=(const DefaultTensorOptions&) = default;
```

Comment:
I always get nervous about this, because this ONLY applies if I have exactly zero constructors, and if I add one, I have to add back the default constructor.